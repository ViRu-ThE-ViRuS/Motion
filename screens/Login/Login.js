import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
    Image,
} from 'react-native';

import { FacebookLoginButton, GoogleLoginButton, EmailLoginButton } from './../../components/Button/LoginButton';

export default class Login extends Component {
    static navigationOptions = {
        header: null,
    }

    navigateToMain() {
        this.props.navigation.navigate('main');
    }

    render() {
        return (
          <View style={styles.container}>
            <View style={styles.center}>
              <Image
                        source={require('./../res/mlogo.png')}
                        style={styles.image}
                        resizeMode="contain"
                    />
              <Text style={styles.subHeadingPrimary}>Login With:</Text>
            </View>

            <View style={styles.buttons}>
              <FacebookLoginButton style={styles.mainButton} onPress={() => this.navigateToMain()} />
              <GoogleLoginButton style={styles.mainButton} onPress={() => this.navigateToMain()} />
              <EmailLoginButton style={styles.mainButton} onPress={() => this.navigateToMain()} />
            </View>
          </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'space-between',
        flex: 1,
    },
    subHeadingPrimary: {
        fontSize: 20,
        color: '#222',
        fontFamily: 'SourceSansPro_Light',
        textAlign: 'center',
        marginLeft: 15,
        marginRight: 15,
        marginTop: 15,
    },
    buttons: {
        alignSelf: 'stretch',
        marginBottom: 45,
    },
    center: {
        marginTop: 150,
        justifyContent: 'center',
        alignItems: 'center',
    },
    image: {
        height: 150,
        width: 150,
    },
});
