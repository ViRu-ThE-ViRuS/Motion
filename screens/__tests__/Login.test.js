import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Login from './../Login/Login';

beforeEach(() => {
    global.fetch = jest.fn((props) => ({
        text: jest.fn(),
    }));
});

it('Login screen renders correctly', () => {
    const mockNavigation = {
        navigate: jest.fn((route) => { console.log(route); }),
    };

    const tree = renderer.create(
      <Login navigation={mockNavigation} />
    );

    expect(tree).toMatchSnapshot();
});
