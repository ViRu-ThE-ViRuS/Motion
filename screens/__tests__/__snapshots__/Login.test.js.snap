// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 150,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../screens/res/mlogo.png",
        }
      }
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontFamily": "SourceSansPro_Light",
          "fontSize": 20,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Login With:
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "marginBottom": 45,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#3b5998",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "SourceSansPro_Regular",
              "fontSize": 15,
            }
          }
        >
          Facebook
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#d34836",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "SourceSansPro_Regular",
              "fontSize": 15,
            }
          }
        >
          Google
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#5C7D87",
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontFamily": "SourceSansPro_Regular",
              "fontSize": 15,
            }
          }
        >
          Email
        </Text>
      </View>
    </View>
  </View>
</View>
`;
